binary_search_tree/tree.py

```diff
@@ -7,42 +7,127 @@ def __init__(self, key, val = None):
         self.value = val
         self.left = None
         self.right = None
-        
 
+    
+    def dict(self):
+        return {"key": self.key, "value": self.value}
 
 class Tree:
     def __init__(self):
         self.root = None
 
+    def add_helper(self, current_node, new_node):
+        if current_node == None:
+            return new_node
+
     # Time Complexity: 
     # Space Complexity: 
     def add(self, key, value = None):
-        pass
+        node = TreeNode(key, value)
+        if self.root == None:
+            self.root = node
+            return
+
+        previous = None
+        current = self.root
+        while current != None:
+            previous = current
+            if key < current.key:
+                current = current.left
+            else:
+                current = current.right
+        if key < previous.key:
+            previous.left = node
+        else:
+            previous.right = node
+                
 
     # Time Complexity: 
     # Space Complexity: 
     def find(self, key):
-        pass
+        if self.root == None:
+            return None
+        current = self.root
+        while current != None:
+            if current.key == key:
+                return current.value
+            elif current.key < key:
+                current = current.right
+            else:
+                current = current.left
+
+
+    def find_helper(self, current, key):
+        if current.key == key:
+            return current.value
+        if current.key > key:
+            return self.find_helper(current.left, key)
+        else:
+            return None
```

Comment:
You're not checking the right side of the tree with this (incomplete?) function.